adapters/repos/db/shard_async.go

```diff
@@ -50,51 +82,57 @@ func (s *Shard) PreloadQueue(targetVector string) error {
 		return nil
 	}
 
-	// load non-indexed vectors and add them to the queue
-	checkpoint, exists, err := s.indexCheckpoints.Get(s.ID(), targetVector)
-	if err != nil {
-		return errors.Wrap(err, "get last indexed id")
-	}
-	if !exists {
-		return nil
-	}
-
-	defer func() {
-		q.metrics.Preload(start, counter)
-	}()
-
 	ctx := context.Background()
 
 	maxDocID := s.Counter().Get()
 
-	err = s.iterateOnLSMVectors(ctx, checkpoint, targetVector, func(id uint64, vector []float32) error {
+	var batch []common.VectorRecord
+	err = s.iterateOnLSMVectors(ctx, from, targetVector, func(id uint64, vector []float32) error {
 		if vectorIndex.ContainsNode(id) {
 			return nil
 		}
 		if len(vector) == 0 {
 			return nil
 		}
 
-		desc := vectorDescriptor{
-			id:     id,
-			vector: vector,
+		rec := common.VectorRecord{
+			ID:     id,
+			Vector: vector,
 		}
-
 		counter++
-		return q.Push(ctx, desc)
+
+		batch = append(batch, rec)
+
+		if len(batch) < 1000 {
```

Comment:
hmm, maybe we could make a const out of this number? tis is just an internal batch size for shard_async? I mean it should not be adjusted in anyways and value 1000 is the right value, yes?

Reply:
yeah it's just an arbitrary value for this function only, any value not too big and not too small works too. Happy to create a const for it